### ISSUE TITLE
lua error when opening document on another computer

### ISSUE THREAD

**Issue by kpuljek** [davidgiven/wordgrinder#56]
Hey!

So I updated Wordgrinder on my main PC (Ubuntu 16.04) to the latest version, compiled it myself, and wrote quite a large article which I then moved to my laptop (Solus) which also has Wordgrinder 0.7.1 installed.

But, when I try to open my document, I get this error:

Lua error: [string "src/lua/fileio.lua"]:538: attempt to index a nil value (local 'doc')
stack traceback:
	[string "src/lua/fileio.lua"]:538: in function 'loadfromstreamt'
	[string "src/lua/fileio.lua"]:574: in function 'LoadFromStream'
	[string "src/lua/fileio.lua"]:581: in upvalue 'loaddocument'
	[string "src/lua/fileio.lua"]:607: in field 'LoadDocumentSet'
	[string "src/lua/main.lua"]:114: in function 'WordProcessor'
	[string "src/lua/main.lua"]:365: in function 'Main'

Making and saving an entirely new document works, it's just this one that won't open. Inspecting a freshly saved document's header with this one, I see absolutely zero differences. Any ideas?

**Comment by davidgiven** [comment]
Yikes, that's not good. However, fear not --- your data is safe (if it was saved with a new enough version you should just be able to load the .wg file in a text editor; if that doesn't work, tools area available).

I actually just fixed a bug in this area on the very line mentioned. You said that you had compiled it yourself on your laptop --- could you update to the latest head of git and try again, please?

Sorry for the inconvenience!

**Comment by kpuljek** [comment]
Yeah, I know about my data, I just migrated everything and completed in Writer for the time being. 

I recompiled it now on my Ubuntu PC, Solus had 0.7.1 in their repository so I didn't do anything there yet. Ubuntu 16.04 has only 0.6.3 so that's the one I upgraded manually. I will resave the document now and try to open it on my laptop again tomorrow and we'll see how it goes. If I can test anything else, just let me know!

**Comment by davidgiven** [comment]
MAKE BACKUPS.

Er, _cough_, please make a copy of the original non-loading document so that, if it the new version works, we can look to see what's different. And, you know, in case of emergencies...

**Comment by kpuljek** [comment]
Compiled the new version on my Ubuntu machine, re-saved the document, still received the same error on my laptop. Then I compiled the new version on my Solus laptop, and now the re-saved document works, but the original one still doesn't. I tried to diff them, this is all I get:

31c31
< .documents.1.co: 8
---
> .documents.1.co: 7
40a41,42
> .menu.accelerators.^@: "ZM"
> .menu.accelerators.^B: "SB"
41a44
> .menu.accelerators.^C: "EC"
43a47,48
> .menu.accelerators.^DOWN: "ZNP"
> .menu.accelerators.^E: "ZDW"
53d57
< .menu.accelerators.ET: "^X"
54a59
> .menu.accelerators.ET: "^X"
55a61
> .menu.accelerators.^F: "EF"
57a64
> .menu.accelerators.^G: "EG"
58a66,68
> .menu.accelerators.^I: "SI"
> .menu.accelerators.^K: "EN"
> .menu.accelerators.^L: "ECfind"
59a70,73
> .menu.accelerators.^LEFT: "ZWL"
> .menu.accelerators.^M: "ECadd"
> .menu.accelerators.^O: "SO"
> .menu.accelerators.^P: "SP"
60a75
> .menu.accelerators.^PGDN: "ZED"
61a77,79
> .menu.accelerators.^PGUP: "ZBD"
> .menu.accelerators.^Q: "FQ"
> .menu.accelerators.^R: "ER"
62a81,82
> .menu.accelerators.^RIGHT: "ZWR"
> .menu.accelerators.^S: "FS"
64a85
> .menu.accelerators.S^DOWN: "ZSNP"
68a90
> .menu.accelerators.S^LEFT: "ZSWL"
71a94
> .menu.accelerators.S^PGDN: "ZSED"
72a96
> .menu.accelerators.S^PGUP: "ZSBD"
73a98
> .menu.accelerators.S^RIGHT: "ZSWR"
76,80d100
< .menu.accelerators.S^DOWN: "ZSNP"
< .menu.accelerators.S^LEFT: "ZSWL"
< .menu.accelerators.S^PGDN: "ZSED"
< .menu.accelerators.S^PGUP: "ZSBD"
< .menu.accelerators.S^RIGHT: "ZSWR"
81a102
> .menu.accelerators.^U: "SU"
82a104,109
> .menu.accelerators.^UP: "ZPP"
> .menu.accelerators.^V: "EP"
> .menu.accelerators.^W: "ZSW"
> .menu.accelerators.^X: "ET"
> .menu.accelerators.^Y: "Eredo"
> .menu.accelerators.^Z: "Eundo"
116,143c143
< .menu.accelerators.^@: "ZM"
< .menu.accelerators.^B: "SB"
< .menu.accelerators.^C: "EC"
< .menu.accelerators.^DOWN: "ZNP"
< .menu.accelerators.^E: "ZDW"
< .menu.accelerators.^F: "EF"
< .menu.accelerators.^G: "EG"
< .menu.accelerators.^I: "SI"
< .menu.accelerators.^K: "EN"
< .menu.accelerators.^L: "ECfind"
< .menu.accelerators.^LEFT: "ZWL"
< .menu.accelerators.^M: "ECadd"
< .menu.accelerators.^O: "SO"
< .menu.accelerators.^P: "SP"
< .menu.accelerators.^PGDN: "ZED"
< .menu.accelerators.^PGUP: "ZBD"
< .menu.accelerators.^Q: "FQ"
< .menu.accelerators.^R: "ER"
< .menu.accelerators.^RIGHT: "ZWR"
< .menu.accelerators.^S: "FS"
< .menu.accelerators.^U: "SU"
< .menu.accelerators.^UP: "ZPP"
< .menu.accelerators.^V: "EP"
< .menu.accelerators.^W: "ZSW"
< .menu.accelerators.^X: "ET"
< .menu.accelerators.^Y: "Eredo"
< .menu.accelerators.^Z: "Eundo"
< .name: "/home/kpuljek/Desktop/Vidi-Android-Gaming.wg"
---
> .name: "/home/kpuljek/Desktop/Vidi-Android-Gaming (New).wg"
298d297
<

**Comment by kpuljek** [comment]
I realized I skipped over one question and edited my answer, but in the meantime you repeated your message so I'll repeat mine (and expand) too! 

I don't have the old version of wordgrinder anymore, only the newest version on both computers, manually compiled.

The old document (created in wordgrinder available in 16.04 repositories) works on the computer it was made on even in the new version of wordgrinder, doesn't work on the laptop in the new version of wordgrinder nor the one that was available in solus repositories. 

The one I opened in the new version of WG on the original pc and resaved works on both. And the only differences are what I posted today, name and menu accelerators.

**Comment by davidgiven** [comment]
Grr, my wifi is playing up (<500 bytes per second sometimes)...

In the _old_ document, could you load the `.wg` file in a text editor and send me all lines beginning with a `#`? I think I know what's happened here.

However, I'm confused that the two systems you've got, both with the new version, are behaving differently with regard to loading the old document. That shouldn't be possible. Can you double check that they both have the same git revision?

**Comment by kpuljek** [comment]
Well, to be fair, I have no idea what's happening now. 

So, I made the original file and synced it to my laptop via Nextcloud. Copied it to my laptop's desktop. That file didn't work, so I updated Wordgrinder on my main PC, resaved the document in the new version, and that didn't work on my laptop either. Then I updated WG on my laptop (even though the version was the same, 0.7.1) and the file worked. But the original still didn't. Now I tried opening up that originally synced file to my main PC and it doesn't work there either. Even though it worked when I made it, saved it and it got synced over and copied. 

I'm now lost in all the revisions, but I have noticed that the one I originally synced over and copied to my desktop had 2 documents in the set, but only one .documents section in the header, I copied it from the resaved file (which had it) and now it works on both computers. So it appears that the original created in 0.6.3 didn't have the .documents.2.* section, even though it contained the main document and the notes section. How I opened it on my PC in 0.7.1 and re-saved, I don't know. Unless opening its copy on my laptop somehow changed it and made it unfixable by the final git version installed later. Mixing so many versions of software is probably not a good idea.

It started working when I added this:

.documents.2.co: 5
.documents.2.cp: 2
.documents.2.cw: 14
.documents.2.margin: 0
.documents.2.name: "Notes"
.documents.2.sticky_selection: false
.documents.2.viewmode: 1
.documents.2.wordcount: 32

This is such a mess now! Sorry, should probably be closed.
